### PR TITLE
feat: github compliance scanner

### PR DESCRIPTION
**Pull request by CalvinRodo**
# Summary | Résumé

This PR adds a scanner that will scan github for compliance with organizational defined best-practices.

This scanner is triggered by an SNS topic.

This scanner has permission to write to an s3 bucket a json report of the configuration of the app, so we can determine if it's compliant with CDS Policies.

### CODE REVIEW

File: bin/scanner_github_test.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+echo 
+#curl "http://scanner-github:8080/2015-03-31/functions/function/invocations" -d '{ }'
```
Comment (patheard) [inline bin/scanner_github_test.sh:4]: nit: remove commented code

File: terragrunt/aws/scanners/github/lambda.tf [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    effect = "Allow"
+
+    actions = [
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents"
+    ]
+
+    resources = [
+      "*"
```
Comment (patheard) [inline terragrunt/aws/scanners/github/lambda.tf:30]: Could this be scoped down to just the Lambda log group?

File: terragrunt/aws/scanners/github/lambda.tf [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    actions = [
+      "s3:ListBucket",
+      "s3:ListBucketVersions",
+      "s3:GetBucketLocation",
+      "s3:Get*",
+      "s3:Put*"
+    ]
+    resources = [
+      "*"
```
Comment (patheard) [inline terragrunt/aws/scanners/github/lambda.tf:59]: Same comment about scoping down if possible.

**Review by patheard — COMMENTED** [review]
Had a few minor comments, but looks good.  I like the simplicity of how the compliance check query is managed.

File: terragrunt/aws/scanners/github/lambda.tf (hunk 2 above)
Comment (CalvinRodo) [reply terragrunt/aws/scanners/github/lambda.tf:30]: This should be in the lambda module.

File: bin/scanner_github_test.sh (hunk 1 above)
Comment (CalvinRodo) [reply bin/scanner_github_test.sh:4]: Done

**Review by patheard — APPROVED** [review]

**Pull request merged**